common/persistence/nosql/nosql_task_store.go

```
@@ -300,7 +300,22 @@ func (t *nosqlTaskStore) CreateTasks(
 			CreatedTime:     now,
 			PartitionConfig: t.Data.PartitionConfig,
 		}
-		ttl := int(t.Data.ScheduleToStartTimeoutSeconds)
+
+		var ttl int
+		// If the Data has a non-zero Expiry value, means that the ask is being re-added to the tasks table.
+		// If that's the case, use the Expiry value to calculate the new TTL value to match history's timeout value.
+		if !t.Data.Expiry.IsZero() {
+			scheduleToStartTimeoutSeconds := int(t.Data.Expiry.Sub(now).Seconds())
+
+			if scheduleToStartTimeoutSeconds > 0 {
+				ttl = scheduleToStartTimeoutSeconds
+			} else {
+				continue
```

this is not supposed to happen, right? if it does happen we will silently lose some tasks. Either log a warning here or return such tasks in the response to be logged at caller side

---

If for some reason the task didn't complete successfully and it has exceeded the TTL (which matches history timeout) then this will happen. My understanding here is that if the task was not processed, for example, we would also silently lose it. And then history would notice because of the timeout and send it to matching again. Does it make sense?